documents/genericdoc/service.go

```diff
@@ -22,6 +22,31 @@ import (
 	logging "github.com/ipfs/go-log"
 )
 
+// Service provides an interface for generic document methods
+type Service interface {
```

Comment:
We can't use the 'documents.Service' interface.

Speaking in set-theory the need common methods (document.Service) are not identical with the method which can be implemented in a generic way.